kratos/python_scripts/apply_periodic_boundary_condition_process.py

```diff
@@ -0,0 +1,120 @@
+            "interval"                    : [0.0, 1e30],
+            "transformation_settings"     : {
+                                                "rotation_settings":{
+                                                    "center":[0,0,0],
+                                                    "axis_of_rotation":[0.0,0.0,0.0],
+                                                    "angle_degree":0.0
+                                                },
+                                                "translation_settings":{
+                                                    "dir_of_translation":[0.0,0.0,0.0],
+                                                    "magnitude":0.0
+                                                }
+                                            },
+            "search_settings"             : {
+                                                "max_results":100000,
+                                                "tolerance": 1E-6
+                                            },
+            "variable_names"              : []
+        }
+        """)
+
+        # Detect "End" as a tag and replace it by a large number
+        if(settings.Has("interval")):
+            if(settings["interval"][1].IsString()):
+                if(settings["interval"][1].GetString() == "End"):
+                    settings["interval"][1].SetDouble(1e30) # = default_settings["interval"][1]
+                else:
+                    raise Exception("The second value of interval can be \"End\" or a number, interval currently:"+settings["interval"].PrettyPrintJsonString())
+
+        settings.ValidateAndAssignDefaults(default_settings)
+
+        if(settings.Has("variable_names")):
+            if(settings["variable_names"].size() == 0):
+                raise Exception("No variables specified to apply periodic boundary conditions.")
+
+        # The computing model part
+        main_model_part_name = settings["model_part_name"].GetString()
+        self.main_model_part = Model[main_model_part_name]
+        computing_model_part_name = settings["computing_model_part_name"].GetString()
+        if(computing_model_part_name != ""):
+            self.computing_model_part = Model[main_model_part_name+"."+computing_model_part_name]
+        else:
+            self.computing_model_part = Model[main_model_part_name]
+
+        # Assign this here since it will change the "interval" prior to validation
+        self.interval = KratosMultiphysics.IntervalUtility(settings)
+
+        # Create the process
+        periodic_parameters = KratosMultiphysics.Parameters("""{}""")
+        periodic_parameters.AddValue("variable_names", settings["variable_names"])
+        periodic_parameters.AddValue("transformation_settings", settings["transformation_settings"])
+        periodic_parameters.AddValue("search_settings", settings["search_settings"])
+
+
+        master_model_part_name = main_model_part_name+"."+settings["first_model_part_name"].GetString()
```

Comment:
why do you call them first/second in the input? Why not also calling them master/slave there?

Reply:
The reason (at least in the beginning) is that any of those two can be used as master and the other as slave. In this implementation I decided to use the first as Master and the second as Slave. 
Also I am not sure if it is nice to have a mention of master and slave in the `periodic boundary conditons` which has nothing to do how the master and slave are chosen.